mozilla_schema_generator/main_ping.py

```diff
@@ -31,17 +31,46 @@ def get_schema(self):
         return self._update_env(schema)
 
     def _update_env(self, schema):
-        user_pref_elem = {
+        integer = {"type": "integer"}
+        string = {"type": "string"}
+        string_map = {
+            "type": "object",
+            "additionalProperties": string,
+        }
+        nested_string_map = {
+            "type": "object",
+            "additionalProperties": string_map,
+        }
+        map_without_values = {
             "type": "object",
             "additionalProperties": {
-                "type": "string"
-            }
+                "type": "object",
+            },
         }
 
+        active_addons = prepend_properties(("environment", "addons", "activeAddons")) + ("additionalProperties", "properties")
```

Comment:
It's unfortunate that we have to special case the logic here, because only the main ping and it's variants will receive these schema updates.